smart-contracts/contracts-common/concordium-contracts-common/src/types.rs

```diff
@@ -794,14 +802,19 @@ impl str::FromStr for Timestamp {
 }
 
 #[cfg(feature = "derive-serde")]
-/// The display implementation displays the timestamp according to RFC3339
-/// format in the UTC time zone.
+/// The display implementation tries to display the timestamp according to
+/// RFC3339 format in the UTC time zone.
+/// If parsing to a [`chrono::DateTime<Utc>`] fails [`Timestamp`] milliseconds,
+/// [`u64`], is returned directly as a string.
```

Comment:
I don't understand this sentence, something seems off about it.

Reply:
Better now?